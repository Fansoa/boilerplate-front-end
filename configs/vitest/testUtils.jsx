import { ThemeProvider } from "@emotion/react";
import theme from "../../src/theme";
import PropTypes from "prop-types";

export const IntegrationProvider = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

IntegrationProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
