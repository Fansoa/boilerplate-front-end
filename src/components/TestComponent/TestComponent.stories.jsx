import TestComponent from ".";

export default {
  title: "TestComponent",
  component: TestComponent,
};

export const TestComponentStory = {
  variant: "bleu",
};
