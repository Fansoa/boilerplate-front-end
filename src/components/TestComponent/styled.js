import styled from "@emotion/styled";

export const Container = styled.div`
  ${({ theme, variant }) => `
    background-color: ${theme.testComponentTheme[variant].backgroundColor}
  `}
`;
