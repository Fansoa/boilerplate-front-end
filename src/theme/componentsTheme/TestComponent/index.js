import global from "../../global";

const testComponentTheme = {
  bleu: {
    backgroundColor: global.colors.blue,
  },
  rouge: {
    backgroundColor: global.colors.red,
  },
};

export default testComponentTheme;
