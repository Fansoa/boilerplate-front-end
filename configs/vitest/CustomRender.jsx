import { render } from "@testing-library/react";
import { IntegrationProvider } from "./testUtils";

const renderWithIntegrationProvider = (children) =>
  render(<IntegrationProvider>{children}</IntegrationProvider>);

export default renderWithIntegrationProvider;
