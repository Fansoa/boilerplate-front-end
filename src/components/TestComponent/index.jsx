import PropTypes from "prop-types";
import { useState } from "react";
import { Container } from "./styled";

const TestComponent = ({ variant }) => {
  const [isClicked, setIsClicked] = useState(false);

  return (
    <Container variant={variant}>
      <p>TestComponent</p>
      <button
        onClick={() => {
          setIsClicked((curr) => !curr);
        }}
      >
        Bouton
      </button>
      {isClicked && <p>isClicked</p>}
    </Container>
  );
};

TestComponent.propTypes = {
  variant: PropTypes.oneOf(["bleu", "rouge"]),
};

TestComponent.defaultProps = {
  variant: "bleu",
};

export default TestComponent;
