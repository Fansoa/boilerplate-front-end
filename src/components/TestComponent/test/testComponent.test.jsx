import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestComponent from "..";
import { beforeEach, expect } from "vitest";
import renderWithIntegrationProvider from "../../../../configs/vitest/CustomRender";

describe("TestComponent", () => {
  let container;

  beforeEach(() => {
    container = renderWithIntegrationProvider(<TestComponent />).container;
  });

  it("Label is displayed", () => {
    const label = screen.getByText(/testcomponent/i);
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("isClicked is displayed when button is clicked", async () => {
    const button = screen.getByRole("button", {
      name: /common:display.the.text/i,
    });
    await userEvent.click(button);
    const isClicked = screen.getByText(/lorem ipsum/i);
    expect(isClicked).toBeInTheDocument();
    await userEvent.click(button);
    expect(isClicked).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
