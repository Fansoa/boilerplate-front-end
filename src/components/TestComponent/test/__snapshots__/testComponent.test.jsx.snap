// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TestComponent > Label is displayed 1`] = `
.emotion-0 {
  background-color: blue;
}

<div>
  <div
    class="emotion-0"
  >
    <p>
      TestComponent
    </p>
    <button>
      Bouton
    </button>
  </div>
</div>
`;

exports[`TestComponent > isClicked is displayed when button is clicked 1`] = `
.emotion-0 {
  background-color: blue;
}

<div>
  <div
    class="emotion-0"
  >
    <p>
      TestComponent
    </p>
    <button>
      Bouton
    </button>
  </div>
</div>
`;
